app/controllers/staff/actuals_controller.rb

```diff
@@ -22,45 +24,45 @@ def new
   def create
     @activity = activity
     authorize @activity
+    @actual = ActualForm.new(actual_params.merge({parent_activity: @activity}))
 
-    result = CreateActual.new(activity: @activity)
-      .call(attributes: actual_params)
-    @actual = result.object
+    actual_created = @actual.valid? && CreateActual.new(
+      activity: @activity,
+      user: current_user,
+      report: Report.editable_for_activity(@activity)
+    ).call(attributes: @actual.attributes).success?
 
-    if result.success?
-      flash[:notice] = t("action.actual.create.success")
-      redirect_to organisation_activity_path(@activity.organisation, @activity)
-    else
-      render :new
-    end
+    return render :new unless actual_created
+
+    flash[:notice] = t("action.actual.create.success")
+    redirect_to organisation_activity_path(@activity.organisation, @activity)
```

Comment:
Using a `return unless` here reads funny to me. Why not:

```ruby
if actual_created
  flash[:notice] = t("action.actual.create.success")
  redirect_to organisation_activity_path(@activity.organisation, @activity)
else
  render :new
end

Reply:
The succinctness seemed to help the readability of the controller, if it doesn't, we can expand it like you suggested.

Reply:
I personally find the "defending return" (there's probably a proper name for it) is better if you're going to be wrapping a lot of LOC or complicated logic inside the `else`, but I find `if... else` easier to read and follow if its just a couple of lines of code like here.